python/GafferSceneUITest/SceneViewTest.py

```diff
@@ -242,5 +242,60 @@ def getViewCameraTransform() :
 		# ideal world we'll fix this, but it's unrelated to what we're testing here.
 		window.removeChild( viewer )
 
+	def testFrame( self ) :
+
+		script = Gaffer.ScriptNode()
+
+		script["Sphere"] = GafferScene.Sphere()
+		script["Sphere1"] = GafferScene.Sphere()
+		script["Sphere"]["transform"]["translate"].setValue( IECore.V3f( -10, 0, 0 ) )
+		script["Sphere1"]["transform"]["translate"].setValue( IECore.V3f( 10, 0, 0 ) )
+
+		script["Group"] = GafferScene.Group()
+		script["Group"]["in"][0].setInput( script["Sphere"]["out"] )
+		script["Group"]["in"][1].setInput( script["Sphere1"]["out"] )
+
+		view = GafferUI.View.create( script["Group"]["out"] )
+		self.assertTrue( isinstance( view, GafferSceneUI.SceneView ) )
+		self.assertTrue( view["in"].getInput().isSame( script["Group"]["out"] ) )
+
+		def cameraContains( scene, focusPath, objectPath = None ) :
+
+			if objectPath is None :
+				objectPath = focusPath
+
+			camera = view.viewportGadget().getCamera()
```

Comment:
I _think_ it works because I'm explicitly testing only with very specifically oriented camera and objects.

But your idea sounds better anyways.